### ISSUE TITLE
Some questions about multi label image classification in MXNet.

### ISSUE THREAD

**Issue by wangg12** [dmlc/mxnet#945]
Hi,
     I want to perform multi label image classification for my own dataset with mxnet. Here are some questions bother me.

* How to prepare the recordIO database or the index file?
According to [Extension: Mutliple Labels for a Single Image](https://github.com/dmlc/mxnet/blob/a6b4baf9824bb3f0bfb8ec804d333913b3bbc0c8/doc/python/io.md) , the index file should have a fixed width of label field. What if my labels are not always the same width which is the common occasion. And will the labels of a single image be treated as a vector like (1,0,0,...,1,...0,...,1)? Or how can I prepare a binary vector label for the image data?

* What loss function should I assign to the training phase? Will the `SoftmaxOutput` deal with multi-label? Or should I implement a `sum of squares` like loss function?

* What evaluation criteria should I choose for the test or val phase? The accuracy metric is for single label.

Thanks!

**Comment by piiswrong** [comment]
you can refer to this example https://github.com/dmlc/mxnet/pull/940

**Comment by yuconglin** [comment]
@piiswrong did you mean issue #940, I cannot find it.

**Comment by yuconglin** [comment]
@wangg12 did you figure out? Thanks a lot in advance.
